src/main/java/org/cactoos/io/AtomicFile.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+
+/**
+ * A file that can only be written to in an atomic way.
+ *
+ * <p>
+ * There is no thread-safety guarantee.
+ *
+ * @author Ashton Hogan (https://twitter.com/TheAshtonHogan)
+ * @version $Id$
+ * @since 0.49.2
+ */
+public final class AtomicFile extends File {
+
+    private static final long serialVersionUID = -276474606113419118L;
+
+    /**
+     * Ctor.
+     *
+     * @param pathname that should be written to
+     */
+    public AtomicFile(final String pathname) {
+        super(pathname);
+    }
+
+    public synchronized void rewrite(final String string, final Charset charset) throws IOException {
```

Comment:
@ashtonhogan Why "rewrite"? Seems like a strange name. Also, the name "string" for the argument is too broad, how about "content"?

Reply:
Will rename to "content".

Doesn't "rewrite" make sense because it will clear the file and write it all again as opposed to merely appending to the end?

Reply:
Why clear and write all again instead of appending? The original ticket says:

>It would be great to be able to write to a file in a fail-safe way so that if a write operation is interrupted, the data is not lost

That doesn't necessarily mean that there'll be no adding to the file, right? Or? 

Anyway, even if you want to overwrite the contents rather than appending, then ``overwrite(...)`` is clearly a better name; and of course whatever the name, the methods needs a JavaDoc! :)

Reply:
@amihaiemil I will rename to `overwrite` and also introduce a vanilla `write`

Reply:
@ashtonhogan You can leave the ``write`` request in a todo (puzzle), let's see about this for now :))